openhands-sdk/openhands/sdk/context/skills/skill.py

```diff
@@ -366,3 +367,190 @@ def load_user_skills() -> list[Skill]:
         f"Loaded {len(all_skills)} user skills: {[s.name for s in all_skills]}"
     )
     return all_skills
+
+
+# Public skills repository configuration
+PUBLIC_SKILLS_REPO = "https://github.com/OpenHands/skills"
+PUBLIC_SKILLS_BRANCH = "main"
+
+
+def _get_skills_cache_dir() -> Path:
+    """Get the local cache directory for public skills repository.
+
+    Returns:
+        Path to the skills cache directory (~/.openhands/skills-cache/).
+    """
+    cache_dir = Path.home() / ".openhands" / "skills-cache"
+    cache_dir.mkdir(parents=True, exist_ok=True)
+    return cache_dir
+
+
+def _update_skills_repository(
```

Comment:
@OpenHands please simplify this function, this is way to complex. We already hard code the URL so no need to be so defensive about repo path